File: InterstitialAdExample/ObjectiveCImplementation/InterstitialAdExample/InterstitialAdExample/ViewController.m
//
//  ViewController.m
//  InterstitialAdExample
//
//  Created by Rishabh Tripathi on 13/11/24.
//

#import "ViewController.h"
#import "ApMobileSDK/ApMobileSDK.h"

@interface ViewController () <ApInterstitialListener>

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
}
- (IBAction)showAdBtnTapped:(UIButton *)sender {
    [ApAppSDK showInterstitialAdWithViewController:self apPlacementID:@"testPlacement" apInterstitialListener:self];
}
- (void)onComplete {
    NSLog(@"Ad Complete");
}
- (void)onAdDismissed {
    NSLog(@"Ad Dismissed");
}
@end

File: InterstitialAdExample/SwiftImplementation/InterstitialAdExample/ViewController.swift
//
//  ViewController.swift
//  InterstitialAdExample
//
//  Created by Rishabh Tripathi on 04/11/24.
//

import UIKit
import ApMobileSDK

class ViewController: UIViewController, ApInterstitialListener {
    
    func onComplete() {
        print("Interstitial Ad Completed")
    }
    func onAdDismissed() {
        print("Interstitial Ad Dismissed")
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        
    }
    
    @IBAction func showAdTapped(_ sender: UIButton) {
        ApAppSDK.showInterstitialAd(viewController: self, apPlacementID: "testPlacement", apInterstitialListener: self)
    }
}

